sdk/search/azure-search/src/main/java/com/azure/search/documents/util/RegexFlagsUtils

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.documents.util;
+
+import com.azure.search.documents.models.RegexFlags;
+import java.util.Arrays;
+
+/**
+ * Utility class for {@link RegexFlags} complex operations.
+ */
+public final class RegexFlagsUtils {
+
+    /**
+     * Combines a list of {@link RegexFlags} into one separated by "|".
+     * E.g. {@code "CANON_EQ|CASE_INSENSITIVE|COMMENTS"}
+     *
+     * @param regexFlags A list a {@link RegexFlags} to combine.
+     * @return New combined {@link RegexFlags} separated by vertical bar.
+     */
+    public static RegexFlags combineRegexFlags(RegexFlags... regexFlags) {
```

Comment:
Would the service return a combined regex flag that a user may want to split up? If so should we offer a `splitRegexFlags` class that has this API shape:

```java
public static RegexFlags[] splitRegexFlags(RegexFlags combinedRegexFlag)
```

Reply:
Nobody has asked for this, but the scenario makes sense. It seems like one more reason to just model the appropriate properties as lists in the first place.

Reply:
This is nice to have feature and addon. I would prefer to have it after GA

Reply:
@sima-zhu If you model flags as string collections, you'll need to implement the splitting logic for deserialization anyway. This should be done sooner rather than later.